aiida_vasp/calcs/workchains/converge.py

```diff
@@ -0,0 +1,1369 @@
+                cls.store_conv,
+            ),
+            cls.init_converged,
+            cls.init_next_workchain,
+            cls.run_next_workchain,
+            cls.verify_next_workchain,
+            cls.results,
+            cls.finalize
+        )  # yapf: disable
+
+        spec.output('output_parameters', valid_type=get_data_class('parameter'))
+        spec.output('remote_folder', valid_type=get_data_class('remote'))
+        spec.output('retrieved', valid_type=get_data_class('folder'))
+        spec.output('output_structure', valid_type=get_data_class('structure'), required=False)
+        spec.output('output_structure_relaxed', valid_type=get_data_class('structure'), required=False)
+        spec.output('output_convergence_data', valid_type=get_data_class('array'), required=False)
+
+    def initialize(self):
+        """Initialize."""
+        self._init_context()
+        self._init_inputs()
+        self._init_conv()
+
+        return
+
+    def _init_inputs(self):
+        """Initialize the inputs."""
+        self.ctx.inputs = init_input(self.inputs)
+
+        return
+
+    def _init_context(self):
+        """Initialize context variables that are used during the logical flow of the BaseRestartWorkChain."""
+        self._init_standard_context()
+        self._init_converge_context()
+
+        return
+
+    def _init_standard_context(self):
+        """Initialize the standard content of context."""
+
+        self.ctx.workchains = []
+        self.ctx.pw_workchains = []
+        self.ctx.kpoints_workchains = []
+        self.ctx.running_pw = False
+        self.ctx.running_kpoints = False
+        self.ctx.replace_nodes = True
+
+        return
+
+    def _init_converge_context(self):
+        """Initialize the converge part of the context."""
+        self.ctx.converge = AttributeDict()
+        self.ctx.converge.settings = AttributeDict()
+        self._init_pw_context()
+        self._init_kpoints_context()
+        self._init_param_context()
+        self._check_context()
+
+        return
+
+    def _init_pw_context(self):
+        """
+        Initialize plane wave cutoff variables and store in context.
+
+        Parameters
+        ----------
+        encut : float, optional
+            The plane wave cutoff in eV. If not set, convergence tests on this
+            parameter are intitiated.
+        encut_start : float, optional
+            The start value of `encut` during convergence tests.
+        encut_step : float, optional
+            The `encut` step size in eV.
+        encut_samples : int
+            The number of `encut` steps to perform.
+        """
+        converge_dict = self.ctx.converge.settings
+        converge_dict['encut'] = None
+        converge_dict['encut_start'] = None
+        converge_dict['encut_step'] = None
+        converge_dict['encut_samples'] = None
+        if 'settings' in self.inputs:
+            settings = self.inputs.settings.get_dict()
```

Comment:
Could be simplified quite a bit by using default values for `dict.get(name, default)`:

```
converge = self.inputs.get('settings', {}).get('converge', {})
converge_dict.encut = converge.get('pw_energy_cutoff', None)
converge_dict.encut_start = ...
...
```
This avoids having to explicitly initialize everything with `None`

Reply:
Yes, this was simply copy and paste from our previous scripts.

Reply:
Which again was based on some bash stuff...

Reply:
Same as the other suggestions, I will make an issue about cleaning the workchain. Pylint also complaints now.